url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/common/RendererConfiguration.java

```diff
@@ -0,0 +1,365 @@
+//===========================================================================
+//=-------------------------------------------------------------------------=
+//= Module history:                                                         =
+//= - October 11 2005 - Gabriel Sarmiento & Lina Rojas: Original base       =
+//=                     version                                             =
+//= - November 1 2005 - Oscar Chavarro: Quality check - comments added      =
+//= - August 6 2006 - Oscar Chavarro: toString method added                 =
+//===========================================================================
+
+package vsdk.toolkit.common;
+
+/**
+The RendererConfiguration class is used to indicate some attributes in which a 
+geometry is to be displayed in screen. The RendererConfiguration class is not
+responsible of display any data, it is just an a suggestion on how to draw it.
+
+@todo: Rename to 'RenderConfiguration' or something similar...
+*/
+public class RendererConfiguration extends FundamentalEntity /*implements Comparable <RendererConfiguration>*/ {
+
+    /// Check the general attribute description in superclass Entity.
+    public static final long serialVersionUID = 20060502L;
+
+    public static final int SHADING_TYPE_NOLIGHT  = 0;
+    public static final int SHADING_TYPE_FLAT     = 1;
+    public static final int SHADING_TYPE_GOURAUD  = 2;
+    public static final int SHADING_TYPE_PHONG    = 3;
+
+    // Indicates the type of shading algorithm to apply in shaders. This must
+    // be one of the values defined in the constants SHADING_TYPE_* of this
+    // class.
+    private int shadingType;
+
+    private boolean surfaces;
+    private boolean wires;
+    private boolean boundingVolume;
+    private boolean selectionCorners;
+    private boolean texture;
+    private boolean bumpMap;
+    private boolean points;
+    private boolean normals;
+    private boolean trianglesNormals;
+    private ColorRgb wireColor;
+
+    public int compareTo(RendererConfiguration other)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 64-79 into a separate method.